### PR TITLE
Support new data type: STRING, BLOB, TIMESTAMP, DATE

### PR DESCRIPTION
**Pull request by Cpaulyz**
Support new data type: STRING, BLOB, TIMESTAMP, DATE

https://apache-iotdb.feishu.cn/docx/VHnMdyW6WobOrZx3o46cuFF9nJd

### CODE REVIEW

File: java/tsfile/src/main/java/org/apache/tsfile/utils/DateUtils.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    String day = dateStr.substring(6, 8);
+    return new Date(
+        Integer.parseInt(year) - 1900, Integer.parseInt(month) - 1, Integer.parseInt(day));
+  }
+
+  public static LocalDate parseIntToLocalDate(int date) {
+    String dateStr = String.valueOf(date);
+    String year = dateStr.substring(0, 4);
+    String month = dateStr.substring(4, 6);
+    String day = dateStr.substring(6, 8);
```
Comment (JackieTien97) [inline java/tsfile/src/main/java/org/apache/tsfile/utils/DateUtils.java]: use `/` and `%` to calculate will be better?

File: java/tsfile/src/main/java/org/apache/tsfile/utils/DateUtils.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    return localDate.getYear() * 10000
+        + localDate.getMonthValue() * 100
+        + localDate.getDayOfMonth();
+  }
+
+  public static Date parseIntToDate(int date) {
+    String dateStr = String.valueOf(date);
+    String year = dateStr.substring(0, 4);
+    String month = dateStr.substring(4, 6);
+    String day = dateStr.substring(6, 8);
```
Comment (JackieTien97) [inline java/tsfile/src/main/java/org/apache/tsfile/utils/DateUtils.java]: use / and % to calculate will be better?

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: java/tsfile/src/main/java/org/apache/tsfile/utils/DateUtils.java (hunk 1 above)
Comment (Cpaulyz) [reply java/tsfile/src/main/java/org/apache/tsfile/utils/DateUtils.java]: Done. And I add a UT for it.

File: java/tsfile/src/main/java/org/apache/tsfile/utils/DateUtils.java (hunk 2 above)
Comment (Cpaulyz) [reply java/tsfile/src/main/java/org/apache/tsfile/utils/DateUtils.java]: Done. And I add a UT for it.

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**